src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -208,6 +219,41 @@ await foreach (ConfigurationSetting setting in _client.GetConfigurationSettingsA
             }
         }
 
+        private async Task LoadInitialConfiguration()

```

Comment:
>LoadInitialConfiguration [](start = 27, length = 24)

The method doesn't always LoadInitialConfiguration as it says. Seems like the caller needs to be smarter about when to invoke this or the maybe the name needs attended to. Maybe splitting it into `IsLoadAllowed()` and `LoadInitial` ?